src/components/sidebar/InputContainer.tsx

```diff
@@ -14,7 +16,7 @@ interface IInputProps {
     value?: string|number|Date;
     validatorRule?: string;
     icon?: React.ReactNode;
-    type?: 'text' | 'number' | 'date';
+    type?: inputType;
```

Comment:
What if we make type as a required property?

Reply:
I've tried to simulate the HTML5 input element, where type also is optional. Empty type field will in both this, and HTML 5 result in a text input.